src/wmtk/utils/polygon_mesh_topology_initialization.cpp

```diff
@@ -0,0 +1,477 @@
+#include "polygon_mesh_topology_initialization.h"
+#include <Eigen/Sparse>
+
+namespace wmtk {
+namespace {
+
+// Flatten a list of lists to a single vector
+VectorXl flatten(const std::vector<std::vector<long>>& list_of_lists)
+{
+    // Get size of flattened array
+    long arr_size = 0;
+    for (long i = 0; i < list_of_lists.size(); ++i) {
+        arr_size += list_of_lists[i].size();
+    }
+
+    // Build flattened array
+    VectorXl arr(arr_size);
+    long count = 0;
+    for (long i = 0; i < list_of_lists.size(); ++i) {
+        for (long j = 0; j < list_of_lists[i].size(); ++j) {
+            arr[count] = list_of_lists[i][j];
+            count++;
+        }
+    }
+    return arr;
+}
+
+// Generate a range vector of integers [start, end)
+std::vector<long> range(long start, long end)
+{
+    assert(start <= end);
+
+    std::vector<long> arr(end - start);
+    for (long i = 0; i < end - start; ++i) {
+        arr[i] = start + i;
+    }
+
+    return arr;
+}
+
+// Build a right inverse g for a function f so that f(g(i)) = i
+VectorXl build_right_inverse(Eigen::Ref<const VectorXl> f)
+{
+    long n = 0;
+    for (long i = 0; i < f.size(); ++i) {
+        n = std::max<long>(n, f[i] + 1);
+    }
+    VectorXl g(n);
```

Comment:
why not have VectorXll build_inverse(f, long size) to reduce redundancy between thte two inverse functinos